### PR TITLE
Refactor for better insight with logs

### PR DESCRIPTION
**Pull request by deckarep**
- [x] - adds appropriate logging before/after processing of tasks
- [x] - adds logging for possible error cases
- [x] - adds a low-level metre specific jobGUID for tracking instances of scheduled jobs with a global unique id.

### CODE REVIEW

File: master.go [hunk 1]
```diff
@@ -105,21 +107,45 @@ func (m *Master) ScheduleFromId(ID string) error {
...
-	req := NewScheduleInput(msg)
+	// jobGUID is a low-level tracking guid at the metre level.  Tasks can also have their
+	// own UID for tracking which is specific to the app.
+	jobGUID := uuid.NewV4().String()
+
+	msg := CreateMsg(Request, taskId, uid, jobGUID, "")
+	req := NewScheduleInput(SerializeMsg(msg))
+	m.logScheduleInfo("About to schedule job", msg)
 	m.SchInpChan <- req
+	m.logScheduleInfo("Job scheduled job", msg)
```
Comment (gsquire) [inline master.go]: Perhaps reword this as "Job with GUID %s scheduled".

**Review by gsquire — COMMENTED** [review]
One wording change, but otherwise LGTM.

File: master.go [hunk 2]
```diff
@@ -105,21 +107,45 @@ func (m *Master) ScheduleFromId(ID string) error {
...
+}
+
 // FIXME : Scheduling happens here. Is this the right design ?
 func (m *Master) Schedule(taskId string, uid string) (string, error) {
-	msg := CreateMsg(Request, taskId, uid, "")
-	req := NewScheduleInput(msg)
+	// jobGUID is a low-level tracking guid at the metre level.  Tasks can also have their
+	// own UID for tracking which is specific to the app.
+	jobGUID := uuid.NewV4().String()
+
```
Comment (georgethomas111) [inline master.go:131]: Right now there is jobGUID for each `schedule`. So if a task has `n` records to process.  

If bufferLength is set to `k` for each `schedule` then we will still have `n/k` jobGUID's in the logs to track. 

But I think it's great to have it.

**Review by georgethomas111 — APPROVED** [review]

File: master.go (hunk 1 above)
Comment (deckarep) [reply master.go]: Good catch.
